### DISCUSSION TITLE
Is it possible to hide the sidebar but keep its data?

### DISCUSSION THREAD

**Discussion by Octobug** [vuejs/vitepress#3190] [Q&A] [answered]
Hey developers & contributors,

I know this need may seem peculiar.

The scenario is that I'm building a blog theme based on VitePress, and I want to adopt VitePress's built-in features as much as possible. That is, I need the Prev/Next links generated by VitePress, but I don't want the sidebar being shown because it doesn't look right for my blog theme.

Thanks. 🙏🏻

**Comment by Octobug** [comment]
I guess that's impossible currently. Some workaround: https://github.com/Octobug/blog/blob/main/.vitepress/theme/components/PrevNext.vue

**Comment by Octobug** [comment]
Sidebar can be hidden but it needs to write `sidebar: false` in every md:

https://vitepress.dev/reference/frontmatter-config#sidebar

**Reply by brc-dd** [reply, marked as answer]
If you want to set it on all pages/most of the pages you can do it using transformPageData hook in config. You won’t need to write it on every page.

**Reply by Octobug** [reply]
Cool, I haven't tried this hook yet. Thank you!
